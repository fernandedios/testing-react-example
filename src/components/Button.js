import React from 'react';

const Button = ({ name, handleClick }) => {
  return (
    <button name={name} onClick={handleClick} className="btn btn-primary">{name}</button>
  );
}

export { Button };
