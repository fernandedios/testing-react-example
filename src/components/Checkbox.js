import React, { Component } from 'react';

class Checkbox extends Component {
  constructor(props) {
    super(props);

    this.state = { isChecked: false };
    this.onChange = this.onChange.bind(this);
  }

  onChange() {
    this.setState({ isChecked: !this.state.isChecked });
  }

  render() {
    return (
      <label>
        <input type="checkbox" className="form-control" checked={this.state.isChecked} onChange={this.onChange} />
        {this.state.isChecked ? this.props.labelOn : this.props.labelOff}
      </label>
    );
  }
}

export { Checkbox };
