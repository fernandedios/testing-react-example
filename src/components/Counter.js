import React, { Component } from 'react';

const doIncrement = (prevState) => ({
  counter: prevState.counter + 1
});

const doDecrement = (prevState) => ({
  counter: prevState.counter - 1
});

class Counter extends Component {
  constructor(props) {
    super(props);
    this.state = { counter: 0 };

    this.onIncrement = this.onIncrement.bind(this);
    this.onDecrement = this.onDecrement.bind(this);
  }

  onIncrement() {
    this.setState(doIncrement);
  }

  onDecrement() {
    this.setState(doDecrement);
  }

  render() {
    return (
      <div className="counter">
        <h3>Counter Component</h3>
        <p>{this.state.counter}</p>
        <button onClick={this.onIncrement}>Increment</button>
        <button onClick={this.onDecrement}>Decrement</button>
      </div>
    );
  }
}

export { Counter, doIncrement, doDecrement };
