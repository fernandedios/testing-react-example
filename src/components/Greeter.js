import React from 'react';

const Greeter = ({ name, greetings }) => {
  return <div>{`${greetings} ${name}!`}</div>;
};

Greeter.defaultProps = {
  name: 'John',
  greetings: 'Hello'
};

export { Greeter };
