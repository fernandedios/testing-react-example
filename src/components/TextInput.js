import React from 'react';

const TextInput = ({ type, name, ...others }) => {
  return <input {...others} name={name} id={name} type={type} className="form-control" />;
};

TextInput.defaultProps = { type: 'text' };

export { TextInput };
