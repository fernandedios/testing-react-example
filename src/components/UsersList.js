import React, { Component } from 'react';
import axios from 'axios';

import { API_URL } from '../constants';

class UsersList extends Component {
  constructor(props) {
    super(props);
     this.state = { users: [] };
  }

  componentWillMount() {
    axios.get(`${API_URL}/users`)
      .then((response) => {
        const users = response.data;
        this.setState({ users });
      })
      .catch(err => console.log(err));
  }

  renderList() {
    if (this.state.users.length === 0) {
      return <div className="loading">Loading...</div>
    }

    const lis = this.state.users.map(({ id, name }) => {
      return <li key={id}>{name}</li>;
    });

    return (
      <ul>
        {lis}
      </ul>
    );
  }

  render() {
    return this.renderList();
  }
}

export { UsersList };
