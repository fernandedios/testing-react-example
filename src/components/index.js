export * from './Button';
export * from './UsersList';
export * from './Checkbox';
export * from './TextInput';
export * from './LoginForm';
export * from './Greeter';
