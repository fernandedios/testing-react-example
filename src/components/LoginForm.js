import React, { Component } from 'react';
import { TextInput } from './TextInput';
import { Button } from './Button';

class LoginForm extends Component {
  constructor(props) {
    super(props);

    this.state = { email: '', password: '' };
    this.onChange = this.onChange.bind(this);
  }

  onChange(e) {
    this.setState({
      [e.target.name]: e.target.value
    });
  }

  render() {
    return (
      <form className="login-form">
        <label>Username: </label>
        <TextInput name="email" onChange={this.onChange} />
        <label>Password: </label>
        <TextInput type="password" name="password" onChange={this.onChange} />
        <Button id="login-button" name="Submit" />
      </form>
    )
  }
}

export { LoginForm };
