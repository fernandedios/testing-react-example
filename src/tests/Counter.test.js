import React from 'react';
import { shallow } from 'enzyme';
import ReactTestRenderer from 'react-test-renderer';

import { Counter } from '../components';

describe('<Counter />', () => {
  const mockFunction = jest.fn();

  it('should be defined', () => {
    expect(Counter).toBeDefined();
  });

  it('renders h3 component title', () => {
    const output = shallow(<Counter />);
    expect(output.find('h3').text()).toEqual('Counter Component');
  });

  it('should render counter number initialized to 0', () => {
    const output = shallow(<Counter />);
    expect(output.find('p').text()).toEqual('0');
  });

  it('should increment counter number when increment button is clicked', () => {
    const output = shallow(<Counter />);
    output.setState({ counter: 0 });
    output.find('button').at(0).simulate('click');
    expect(output.state().counter).toEqual(1);
  });

  it('should decrement counter number when decrement button is clicked', () => {
    const output = shallow(<Counter />);
    output.setState({ counter: 0 });
    output.find('button').at(1).simulate('click');
    expect(output.state().counter).toEqual(-1);
  });

  it('should render correctly', () => {
    const output = ReactTestRenderer.create(<Counter />).toJSON();
    expect(output).toMatchSnapshot();
  });
});
