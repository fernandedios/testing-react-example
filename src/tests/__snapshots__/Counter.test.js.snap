// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> should render correctly 1`] = `
<div>
  <h3>
    Counter Component
  </h3>
  <p>
    0
  </p>
  <button
    onClick={[Function]}
  >
    Increment
  </button>
  <button
    onClick={[Function]}
  >
    Decrement
  </button>
</div>
`;
